frontend/apps/reader/modules/readerdictionary.lua

```diff
@@ -435,6 +551,93 @@ function ReaderDictionary:showDict(word, results, box)
     end
 end
 
+function ReaderDictionary:showDownload()
+    local kv_pairs = {}
+    for dummy, dict in ipairs(downloadable_dicts) do
+        table.insert(kv_pairs, {dict.name, "",
+            callback = function()
+                self:downloadDictionaryPrep(dict)
+            end})
+        local lang
+        if dict.lang_in == dict.lang_out then
+            lang = string.format("    %s", dict.lang_in)
+        else
+            lang = string.format("    %s–%s", dict.lang_in, dict.lang_out)
+        end
+        table.insert(kv_pairs, {lang, dict.license .. string.format(_(" (%i entries)"), dict.entries)})
+        table.insert(kv_pairs, "----------------------------")
+    end
+    self.download_window = KeyValuePage:new{
+        title = _("Download dictionaries"),
+        kv_pairs = kv_pairs,
+    }
+    UIManager:show(self.download_window)
+end
+
+function ReaderDictionary:downloadDictionaryPrep(dict)
+    local dummy, filename = util.splitFilePathName(dict.url)
+    local download_location = string.format("%s/%s", self.data_dir, filename)
+
+    local lfs = require("libs/libkoreader-lfs")
+    local continue = false
+    if lfs.attributes(download_location) then
+        UIManager:show(ConfirmBox:new{
+            text =  _("File already exists. Overwrite?"),
+            ok_text =  _("Overwrite"),
+            ok_callback = function()
+                self:downloadDictionary(dict, download_location)
+            end,
+        })
+    else
+        self:downloadDictionary(dict, download_location)
```

Comment:
Can we have a ConfirmBox before download start?
Could be nice to have the size of what will be downloaded, so one can choose small dicts, or see it's too big to download now.

Reply:
Yes, (approximate) size might be more relevant than number of entries. There's a definite correlation between the two anyway.